packages/gatsby/src/redux/nodes.ts

```diff
@@ -441,14 +442,23 @@ function addNodeToFilterCache(
       return
     }
 
-    // This means that either
-    // - The filter resolved to `undefined`, or
-    // - The filter resolved to something other than a primitive
-    // Set the value to `undefined` to mark "path does not (fully) exist"
-    v = undefined
+    if (v instanceof Date) {
```

Comment:
So current behavior is to always compare dates as strings, however in this case it won't actually "normalize" this behavior unless all values are of value Date.

What if we always do full normalization so that the date value is always encoded as ISO string inside the cache? Technically it's not a bug to have dates sort/filter correctly.

Reply:
Isn't that exactly what happens here? If a leaf value is found to be `Date`, convert it to iso string. It will do this for all Date values found. Unless you mean encoded in the `IGatsbyNode` instance. I suppose we can do that but this feels like a much much larger change, way out of scope for this PR.

Reply:
Can be not that hard, we already have `serialize-node` step, so it's probably a matter of adding a date serialization somewhere along those lines:

https://github.com/gatsbyjs/gatsby/blob/67f8266fd326feba6b721e5be42b234b5b1fea6b/packages/gatsby/src/db/sanitize-node.js#L38-L42